cybershell/vulnerability_kb.py

```
@@ -0,0 +1,740 @@
+    product: Optional[str] = None  # e.g., "nginx", "apache", "wordpress"
+    version_spec: Optional[str] = None  # e.g., "<=1.14.0", ">=2.0.0,<2.1.0"
+    
+    def matches_version(self, target_version: str) -> bool:
+        """Check if payload matches target version"""
+        if not self.version_spec or not target_version:
+            return True  # If no version spec, payload applies to all versions
+        
+        try:
+            spec = SpecifierSet(self.version_spec)
+            target_v = version.parse(target_version)
+            return target_v in spec
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)